src/InstrumentationEngine.Api/InstrumentationEngine.idl

```diff
@@ -1551,4 +1551,83 @@ library MicrosoftInstrumentationEngine
         HRESULT JitComplete([in] IMethodJitInfo* pJitInfo);
     };
 
-}
+    [
+        object,
+        uuid(EF0B0C79-08E7-4C3A-A4C5-02A9C9CE8809),
+        pointer_default(unique)
+    ]
+    interface IInstrumentationMethodSetting : IUnknown
+    {
+        // Gets the name of the instrumentation method setting.
+        HRESULT GetName(
+            [out] BSTR* pbstrName
+            );
+
+        // Gets the value of the instrumentation method setting.
+        HRESULT GetValue(
+            [out] BSTR* pbstrValue
+            );
+    };
+
+    [
+        object,
+        uuid(9B03D87E-72F0-4D8E-A4B1-15BCD8227073),
+        pointer_default(unique)
+    ]
+    interface IEnumInstrumentationMethodSettings : IUnknown
+    {
+        HRESULT Next(
+            [in] ULONG celt,
+            [out, size_is(celt), length_is(celt)] IInstrumentationMethodSetting** rgelt,
+            [out] ULONG* pceltFetched
+            );
+
+        HRESULT Skip(
+            [in] ULONG celt
+            );
+
+        HRESULT Reset(void);
+
+        HRESULT Clone(
+            [out]IEnumInstrumentationMethodSettings** ppenum
+            );
```

Comment:
What is the purpose of the Clone function? I understand that it would be to copy the enum to another enum. But would the clone be a deep copy, or a shallow copy? Would it just be the same enum with the index reset? I'm a little confused on why you would want to clone an existing enum rather than just getting another enum from the same object.

Reply:
It creates a new CEnumerator instance with a new CComPtr vector of the same pointers elements from the existing enumerator. See CEnumerator::Clone in src/InstrumentationEngine.Lib/EnumeratorImpl.h.

Every other IEnum* interface in the IDL file uses this pattern, so I'm continuing it here.